src/test/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProviderIT.java

```diff
@@ -18,168 +18,125 @@
-import com.snowflake.kafka.connector.Utils;
 import com.snowflake.kafka.connector.internal.TestUtils;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
-import net.snowflake.ingest.internal.com.github.benmanes.caffeine.cache.LoadingCache;
 import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.mockito.Mockito;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
-@RunWith(Parameterized.class)
 public class StreamingClientProviderIT {
-  private final boolean enableClientOptimization;
-  private final Map<String, String> clientConfig = TestUtils.getConfForStreaming();
 
-  @Parameterized.Parameters(name = "enableClientOptimization: {0}")
-  public static Collection<Object[]> input() {
-    return Arrays.asList(new Object[][] {{true}, {false}});
+  @Test
+  public void getClient_forOptimizationEnabled_returnSameClient() {
+    // given
+    Map<String, String> clientConfig = getClientConfig(true);
+    StreamingClientProvider streamingClientProvider =
+        StreamingClientProvider.getStreamingClientProviderInstance();
+
+    // when
+    SnowflakeStreamingIngestClient client = streamingClientProvider.getClient(clientConfig);
+    SnowflakeStreamingIngestClient client2 = streamingClientProvider.getClient(clientConfig);
+
+    // then
+    Assertions.assertNotNull(client);
+    Assertions.assertNotNull(client2);
+    Assertions.assertEquals(client, client2);
+
+    int clientsRegistered = streamingClientProvider.getRegisteredClients().size();
+    Assertions.assertEquals(1, clientsRegistered);
   }
 
-  public StreamingClientProviderIT(boolean enableClientOptimization) {
-    this.enableClientOptimization = enableClientOptimization;
-    this.clientConfig.put(
-        SnowflakeSinkConnectorConfig.ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG,
-        String.valueOf(this.enableClientOptimization));
+  @Test
+  public void getClient_forOptimizationDisabled_returnDifferentClients() {
+    // given
+    Map<String, String> clientConfig = getClientConfig(false);
+    StreamingClientProvider streamingClientProvider =
+        StreamingClientProvider.getStreamingClientProviderInstance();
+
+    // when
+    SnowflakeStreamingIngestClient client = streamingClientProvider.getClient(clientConfig);
+    SnowflakeStreamingIngestClient client2 = streamingClientProvider.getClient(clientConfig);
+
+    // then
+    Assertions.assertNotNull(client);
+    Assertions.assertNotNull(client2);
+    Assertions.assertNotEquals(client, client2);
+  }
+
+  @Test
+  public void getClient_forInvalidClient_returnValidClient() throws Exception {
```

Comment:
`getClient_forInvalidClient_returnValidClient` sounds a bit odd :)
Is it intended?

Reply:
Yeah, kind of weird. i renamed to `getClient_forInvalidClient_returnNewInstance`